### PR TITLE
Makes multiple adjustments to slipping

### PR DESCRIPTION
**Pull request by Lady-Luck**
This makes some adjustments to the slip durations. Some are the same, some are lower. The formula for the duration is roughly ((x*2)-1) so a 3 lasts ~5 seconds and a 4 lasts ~7 seconds.

The biggest changes were Water, which went from 7 seconds to 5 seconds, Space Lube, which went from 13 seconds to 5 seconds, and the Clown PDA, which went from 15 seconds to 7 seconds (which makes it the same as a banana peel or soap).

Additionally, tiles slipped on space lube went from 4 to 3.

Nothing here is final, everything is up to discussion, so no reason to vote for blanket disapproval just yet.

### CODE REVIEW

**Comment by tigercat2000** [comment]
>Space Lube, which went from 13 seconds to 5 seconds

what
so now space lube is the exact same stun as water :V

**Comment by Lady-Luck** [comment]
No, you can't walk on lube.

**Comment by Necaladun** [comment]
Honestly I like it. Stuns from falling over should not be so damn bad. They should be a minor annoyance, not a powerful combat tactic. Otherwise we'd see cops and the USMC carrying water balloons instead of flashbangs.

**Comment by Lady-Luck** [comment]
Also of note, I did tried a weaken proc of 2, but ~3 seconds just feels too short to do anything useful with in combat.

**Comment by Spacemanspark** [comment]
:+1: Because lube in spray bottles is just terrible.

**Comment by TheDZD** [comment]
The comments probably aren't needed. Coders should at least be smart enough to figure out (grep/the search function are wonderful), or at least ask how slips work.

**Comment by Lady-Luck** [comment]
I like comments. I feel like people will look at a 3 there and simply assume it means 3 seconds, so they won't even think to ask. Just for convenience.

**Comment by Lady-Luck** [comment]
>The comments probably aren't needed. Coders should at least be smart enough to figure out (grep/the search function are wonderful), or at least ask how slips work. Also probably not accurate giving the time in seconds, since stuns are Life() tick-based and as a result can vary, both due to server load and the timing of effects that are not synchronous with Life() ticks.

I tested the durations on the live server, our Life() ticks were pretty close to what I was getting on my private local server.

**Comment by Ar3nn** [comment]
I do like the nerf on space lube, simply because someone lol-lubed the halls once during nuke ops and it completely trashed them because of the long timer. Someone walked by and rolled god-mode to run across the lube, grab my gun and shoot me in the face with it while backpedaling (and still not slipping) because rngeezus.

**Comment by Earthdivine** [comment]
My only regret is we will probably never see the coveted "Lube the floor, watch them slide into space" again. o7

**Comment by Lady-Luck** [comment]
>My only regret is we will probably never see the coveted "Lube the floor, watch them slide into space" again. o7

Why?

**Comment by TheDZD** [comment]
Comment about the commerns is largely due to the fact that it just adds pointless commits, which end up cluttering the commit log even further (though it's already hellish anyways) for something that is entirely unnecessary. Space lube nerf is definitely needed though, something you can cover floors with to completely deny movement across a hallway should not have a stun comparable to a baton. No idea about the rest at the moment.

**Comment by Lady-Luck** [comment]
I honestly meant to have the comments in the first place, but sadly it slipped my mind. I'll keep that in mind in the future.

**Comment by Ssirius** [comment]
:+1: To lube nerf, I don't think other slips need nerfs tho, since all of them have counters and ways to avoid them, walk over water, take the peels/PDA  just walk around them, the PDA is supposed to be special, considering its the only one of its kind per clown.

**Comment by Lady-Luck** [comment]
>To lube nerf, I don't think other slips need nerfs tho, since all of them have counters and ways to avoid them, walk over water, take the peels/PDA just walk around them, the PDA is supposed to be special, considering its the only one of its kind per clown.

The PDA changed was based off my own experiences with clown. Using the PDA to slip people for a lot of time was fucking awesome, but sometimes it lead to it get stolen, and then I'd have to bother security about it. So I decided the best thing to do would be to just make it the same duration as peel, so peel is preferred, but the PDA becomes a backup.

**Comment by Ssirius** [comment]
> The PDA changed was based off my own experiences with clown. Using the PDA to slip people for a lot of time was fucking awesome, but sometimes it lead to it get stolen, and then I'd have to bother security about it. So I decided the best thing to do would be to just make it the same duration as peel, so peel is preferred, but the PDA becomes a backup.

Well that's the thing, if you use it you can lose it, its a special item that can get stolen since its quite strong, the PDA is a last resort slip, a secret special weapon if you will, clown can even grow his own bananas for infinite normal slips.

**Comment by Lady-Luck** [comment]
>Well that's the thing, if you use it you can lose it, its a special item that can get stolen since its quite strong, the PDA is a last resort slip, a secret special weapon if you will, clown can even grow his own bananas for infinite normal slips.

Actually it was a first resort slip in my experience just because of how much stronger it was. By making it the same as the peel, it turns into a last resort.

**Comment by Bigtimetripod** [comment]
Are people forgetting a secret chem that can be made that will dry lube and water from tiles???  Be a robust ops, make a spray full of drying agent and fuck lubers, they have nohting against ye anymore.

**Comment by Lady-Luck** [comment]
>Are people forgetting a secret chem that can be made that will dry lube and water from tiles??? Be a robust ops, make a spray full of drying agent and fuck lubers, they have nohting against ye anymore.

Degreaser is one of the hardest chems in the game to make, and spray bottles are in short enough supply to not want to provide them to Security.

**Comment by Lady-Luck** [comment]
>Are people forgetting a secret chem that can be made that will dry lube and water from tiles??? Be a robust ops, make a spray full of drying agent and fuck lubers, they have nohting against ye anymore.

Degreaser is a huge pain in the ass to make, requiring several lengthy steps to counter a chemical that is just Oxygen, Silicon, and Water. Furthermore, spray bottles are in short enough supply to not want to provide one to even a single Officer, let alone the entire force.

**Comment by Twinmold93** [comment]
Water and space lube slip time shouldn't be the same, and water slip should really only be about as long as it is now. Space lube slip time should be lower than it is now, but not by much, and the clown PDA slip time should be the same as a banana.

**Comment by Lady-Luck** [comment]
>Water and space lube slip time shouldn't be the same, and water slip should really only be about as long as it is now. Space lube slip time should be lower than it is now, but not by much, and the clown PDA slip time should be the same as a banana.

Many admins I talked to felt water and lube should be the same in duration, because of the caveat that you can't walk on it like you can with water. A large majority of Discord agreed.

**Comment by Fox-McCloud** [comment]
I ded, pls nerf.

I swear, people incessantly complain about how hugboxed Paradise has become, but then remove some of the things that actually makes it dangerous and scary. You can't have it both ways, either it's dangerous and scary and sometimes brutal, or it's a hugbox where only the most horrific of things results in death or permanent removal from the round.

I don't like this for that reason.

Secondly, I don't recommend messing with the stun+weaken values, for the simple matter of, not everything that reduces stuns also reduces weakens. In *most* cases, yes, it does, but not always--for that reason, I'm leery, on a coded perspective, of messing with them.

**Comment by Lady-Luck** [comment]
I have never died to a lube sprayer.

**Comment by Lady-Luck** [comment]
>Secondly, I don't recommend messing with the stun+weaken values, for the simple matter of, not everything that reduces stuns also reduces weakens. In most cases, yes, it does, but not always--for that reason, I'm leery, on a coded perspective, of messing with them.

You'll find that this doesn't matter much. In the code, the stun duration was, in almost every case, twice as long as the weaken duration. So unless the stun reduction is more than 50%, it won't matter.

**Comment by Bigtimetripod** [comment]
@lady-luck 

2 parts plasma, 1 part ethanol, 1 part sodium is childplay to mix for drying agent... not complicated at all... 

As Fox sayd, ye seem to try to make SS 13 a game with soft white bouncy walls... do not mess with things that has been set in place since the beginning. If yer thing is to lay yer mark on this community, then try some other PR... this is not the one, game is ment to be harsh... 

Im not trying to be negative, but do not cut all the values in half... 8-10 seconds on space lube stun is enough... its ment to be a industrial lube, not some cheap sex lube.

**Comment by Lady-Luck** [comment]
>drying agent

Why on earth is this not on the wiki? This is the first time I've ever heard of it.

That said, people have been wanting slipping mechanics toned down for a LONG time. Hell, recently we nerfed tabling. It's time lube got a little less ridiculous. Even with this change, it's going to be ridiculously strong.

**Comment by Bigtimetripod** [comment]
Drying agent is a oldchem reagent, its mostly useless, but has a very specific use. Easly forgotten cause its almost mever used unless enemy is hellbent on using slipping.

 Table nerf was something i didnt like... again, basically lets make the tables out of cotton so it stuns for only 2 seconds, not 7... robust as fuck was that move...

**Comment by Lady-Luck** [comment]
New table stun is 3 seconds as opposed to 13.

**Comment by Lady-Luck** [comment]
@Fox-McCloud It now stuns and weakens again.

**Comment by Regen1** [comment]
>I ded, pls nerf.
I swear, people incessantly complain about how hugboxed Paradise has become, but then remove some of the things that actually makes it dangerous and scary. You can't have it both ways, either it's dangerous and scary and sometimes brutal, or it's a hugbox where only the most horrific of things results in death or permanent removal from the round.

KEK

I support this change, space lube being an instant "I win" factor (just like how tasers are) is stupid
:+1:

**Comment by Fox-McCloud** [comment]
Having a stun be some multiple of the weaken isn't very good, for the reason it doesn't allow for as much granular control.

That is to say, if someone wants to have a weaken of 10, but a stun of 1, they can't do that.

The way you have it set up locks it into being a set multiple of whatever weaken is.

This also means that we couldn't make a slip that ONLY stunned you, but didn't weaken you.

**Comment by Lady-Luck** [comment]
If a slip only stunned you but didn't weaken you, it wouldn't be a slip.

**Comment by Lady-Luck** [comment]
Bear in mind that these two effects apply simultaneously. They don't add together.

**Comment by KasparoVy** [comment]
One of my first rounds of SS13 saw me slip on lube outside the bar and end up in space outside escape

I thought it was absolutely hilarious

**Comment by Fox-McCloud** [comment]
You're missing the point here.

If for whatever reason we wanted to make a way for that to happen, it shouldn't be hindered and held up because it's defined as an arbitrary divisor like you have it set up. Modular and granular control over precise numbers is a good thing--having one be based on the other (when there's ZERO reasons too), is bad.

**Comment by Lady-Luck** [comment]
>If for whatever reason we wanted to make a way for that to happen, it shouldn't be hindered and held up because it's defined as an arbitrary divisor like you have it set up. Modular and granular control over precise numbers is a good thing--having one be based on the other (when there's ZERO reasons too), is bad.

Weaken does literally everything stun does when it comes to slipping. I set it up this way so that those with stun reduction but not weaken reduction will be able to get some faster. Furthermore, it was causing confusion not just with myself, but with coders I was asking about the slip proc with. Simplicity works best.

**Comment by Fox-McCloud** [comment]
This isn't an argument. You're literally making the proc less granular and less modular, on the whole. There's no reason for this, *at all*.

**Comment by Lady-Luck** [comment]
I just gave you the reason.

**Comment by tigercat2000** [comment]
stunned in combination with weaken literally does absolutely nothing
this literally means you are getting a much different result than you expect
just make it `Weaken(argValue)` and increase them across the board if you want it to be so

**Comment by Lady-Luck** [comment]
>stunned in combination with weaken literally does absolutely nothing
this literally means you are getting a much different result than you expect
just make it Weaken(argValue) and increase them across the board if you want it to be so

I'm aware. Read the comments here.

**Comment by tigercat2000** [comment]
whatever, your PR to not get merged~

**Comment by Lady-Luck** [comment]
>whatever, your PR to not get merged~

Uh, why?

**Comment by Lady-Luck** [comment]
Had a long talk with tigercat on Discord. He brought up a good point in that porting slips from -tg- will be made more annoying with this change to the slip proc. So I am changing it back but leaving the new values.

File: code/modules/pda/pdas.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/obj/item/device/pda/medical
+f/obj/item/device/pda/medical
```
Comment (Fox-McCloud) [inline code/modules/pda/pdas.dm]: uhhhhhhhhh

Comment (Lady-Luck) [inline code/modules/pda/pdas.dm]: No idea how the hell that happened. Gonna blame the cat.

**Comment by Fox-McCloud** [comment]
Please remove the unnecessary comments.

I'd also recommend leaving the values for everything as they were and consider only nerfing lube.

I'd personally recommend it having a duration of 5 and it doesn't do any damage anymore, but that's me. What do you think @TheDZD?

**Comment by TheDZD** [comment]
Yeah, I don't see much of a point in changing all these values when the end result is ultimately the same for the general case, and in the end just messes with exceptions (anything that reduces stun but not weaken, or vice versa, etc). Wouldn't mind seeing the damage be removed either, although the disable should probably still be a bit longer than water.

**Comment by Lady-Luck** [comment]
Duration of 5 would be 9 seconds... that barely touches the power of lube.

**Comment by tigercat2000** [comment]
comments should only be used to clarify complicated or not-obvious code

this is neither, it sets a value which is reduced every 2 seconds

**Comment by Lady-Luck** [comment]
To someone working with BYOND for the first time, it wasn't anywhere near obvious.

I'll remove the comments. But to take the value of lube only down to 5 still leaves it with the same stun duration as the old table stun - but it comes with just amazing area denial. 13 seconds to 9 seconds is such a nothing change.

**Comment by TheDZD** [comment]
PR is missing a changelog.

**Comment by Fox-McCloud** [comment]
No updates in a quite a while, not even to the changelog. Closing this.

**Comment by Lady-Luck** [comment]
Sorry for dropping the ball. I was ill for several weeks, severe migraines, and I had to go to the emergency room once or twice.

It would be best if someone else did this. I have a hard time handling all the really combative arguing that goes on here, it stresses me out and takes the fun out of it.

**Comment by Ar3nn** [comment]
I'll just say that tackling something like this was bound to get some shots fired. You proposed some pretty severe nerfs to one of the most iconic game mechanics. Try adding some small things, like I've been doing. I screwed up by trying to go balance at first, too, (See my Voltine PR and literally everyone hating the mere thought of IPC getting nicotine, because muh drugs, and literally no other reason.) Hang in there, though!

**Pull request closed without merging**